lib/pbench/test/unit/server/conftest.py

```diff
@@ -736,31 +700,131 @@ def create_drb_user(client, server_config, fake_email_validator):
 
 
 @pytest.fixture()
-def pbench_token(client, server_config, create_drb_user):
-    # Login user to get valid pbench token
-    response = login_user(client, server_config, "drb", generic_password)
-    assert response.status_code == HTTPStatus.OK
-    data = response.json
-    assert data["auth_token"]
-    return data["auth_token"]
+def pbench_admin_token(client, create_admin_user):
+    return generate_token(
+        user=create_admin_user,
+        username=admin_username,
+        pbench_client_roles=["ADMIN"],
+    )
+
+
+@pytest.fixture()
+def pbench_token(client, create_drb_user):
+    """
+    OIDC valid token for the 'drb' user.
+    """
+    return generate_token(username="drb", user=create_drb_user)
+
+
+@pytest.fixture()
+def pbench_token_invalid(client, create_drb_user):
+    """
+    OIDC invalid token for the 'drb' user
+    """
+    return generate_token(username="drb", user=create_drb_user, valid=False)
+
+
+def generate_token(
+    username: str,
+    user: Optional[User] = None,
+    pbench_client_roles: Optional[list[str]] = None,
+    valid: Optional[bool] = True,
+) -> str:
+    """
+    Generates an OIDC JWT token that mimics the real OIDC token obtained
+    from the OIDC complaint client login.
+
+    Args:
+        username: username to include in the token payload
+        user: user attributes will be extracted from the user to include in
+            the token payload, if not provided it will be queried from the User
+            table. It is a responsibility of the consumer to make sure the
+            user exists in the 'User' table until we remove the User table
+            completely.
+        pbench_client_roles: List of any roles to add in the token payload
+        valid: If True, the generated token will be valid for 10 mins.
+               If False, generated token would be invalid and expired
+
+    TODO: When we remove the User table we need to update this functionality's
+        dependance on user table.
+        Refer related Jira issues:
+            https://issues.redhat.com/browse/PBENCH-934,
+            https://issues.redhat.com/browse/PBENCH-900,
+            https://issues.redhat.com/browse/PBENCH-895,
+            https://issues.redhat.com/browse/PBENCH-962
+
+    Returns
+        JWT token string
+    """
+    # Current time to encode in the token payload
+    current_utc = datetime.datetime.now(datetime.timezone.utc)
+
+    # Expired epoch time to use as a token expiry in case the caller
+    # wants an invalid token.
+    EXPIRY = "1667187770"
+
+    if not user:
+        user = User.query(username=username)
```

Comment:
Is `None` an acceptable value for `user`, or should we have an assertion here which fails the test if things are screwed up?

Reply:
No, `None` is definitely not valid. The code will blow up if we can't find a `username` since we're dereferencing the `id` field later; but an `assert` here would be a cleaner failure mode.